ConsoleApp1/Program.cs

```diff
@@ -0,0 +1,118 @@
+                Dictionary<int, Country> countries = new Dictionary<int, Country>();
+                string subline = ";";
+                int startIndex = 0;
+                foreach (string countryElement in countryList)
+                {
+                    int index = countryElement.IndexOf(subline);
+                    Country country = new Country
+                    {
+                        Name = countryElement.Substring(startIndex, index)
+                    };
+                    if (countryElement.Substring(index + 1).ToLower() == "true")
+                    {
+                        country.IsTelenorSupported = true;
+                    }
+                    else
+                    {
+                        country.IsTelenorSupported = false;
+                    }
+                    countries.Add(i + 1, country);
+                    i++;
+                }
+                return countries;
+            }
+            return null;
+        }
+
+        public static Dictionary<int, Country> AddCountryToDictionary(Dictionary<int, Country> dictionary)
+        {
+            bool flag = false;
+            foreach (KeyValuePair<int, Country> kvp in dictionary)
+            {
+                if ((kvp.Value.Name == "Ukraine"))
+                {
+                    flag = true;
+                }
+            }
+            if (flag == true)
+            {
+                return dictionary;
+            }
+            else
+            {
+                Country country = new Country() { 
+                    Name = "Ukraine", 
+                    IsTelenorSupported = true };
+
+                dictionary.Add(dictionary.Count + 1, country);
+                return dictionary;
+            }
+        }
+
+        public static void PrintDictionary(Dictionary<int, Country> dictionary)
+        {
+            Console.WriteLine("\nHere is the Dictionary: ");
+            foreach (KeyValuePair<int, Country> kvp in dictionary)
+            {
+                Console.WriteLine("Key = {0}, Value Name = {1}, Value IsTelenorSupported = {2}", kvp.Key, kvp.Value.Name, kvp.Value.IsTelenorSupported);
+            }
+        }
+
+        public static Dictionary<int, Country> EditIsTelenorSupportedInDictionary(Dictionary<int, Country> dictionary)
+        {
+            foreach (KeyValuePair<int, Country> kvp in dictionary)
+            {
+                if ((kvp.Value.IsTelenorSupported != true) && ((kvp.Value.Name == "Denmark") || (kvp.Value.Name == "Hungary")))
+                {
+                    kvp.Value.IsTelenorSupported = true;
+                    Console.WriteLine("Updates: Key = {0}, Value Name = {1}, Value IsTelenorSupported = {2}", kvp.Key, kvp.Value.Name, kvp.Value.IsTelenorSupported);
+                }
+            }
+        return dictionary;
+        }
+
+        public static List<string> ConvertDictionaryToList(Dictionary<int, Country> dictionary)
```

Comment:
please, use linq here so it's a 1-row action

Reply:
I am not sure I could use linq here as I need to append ";" delimiter, so leaving as is for now

Reply:
that shouldn't be a problem, in this case it is used like this:
return dictionary.Select(entry => entry.Value.Name + ";" + entry.Value.IsTelenorSupported).ToList();